bigtable/bigtable-proxy/src/main/java/com/google/cloud/bigtable/examples/proxy/core/ProxyHandler.java

```diff
@@ -25,15 +26,23 @@
 
 /** A factory pairing of an incoming server call to an outgoing client call. */
 public final class ProxyHandler<ReqT, RespT> implements ServerCallHandler<ReqT, RespT> {
+  private static final Metadata.Key<String> AUTHORIZATION_KEY =
+      Metadata.Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
+
   private final Channel channel;
+  private final CallCredentials callCredentials;
 
-  public ProxyHandler(Channel channel) {
+  public ProxyHandler(Channel channel, CallCredentials callCredentials) {
     this.channel = channel;
+    this.callCredentials = callCredentials;
   }
 
   @Override
   public ServerCall.Listener<ReqT> startCall(ServerCall<ReqT, RespT> serverCall, Metadata headers) {
-    CallOptions callOptions = CallOptions.DEFAULT;
+    // Strip incoming credentials
+    headers.removeAll(AUTHORIZATION_KEY);
```

Comment:
So if someone runs their client with a credential they also need to start the proxy handler with that credential? why not just using the incoming credentials?

Reply:
Thats not quite right. The proxy listening port uses an insecure channel (plaintext) and the customer's client is expected to use emulator mode to connect to the proxy, which disables credentials. The only usecase that I want to support here is the proxy owning the creds